src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java

```diff
@@ -51,15 +54,41 @@
   private final Integer appCommandGsUtil;
   private final File appCommandCredentialFile;
   private final String appCommandOutputFormat;
+  private final int runDevAppServerWaitSeconds;
+  private final WaitingProcessOutputLineListener runDevAppServerWaitListener;
+
+
+  private CloudSdk(Path sdkPath, String appCommandMetricsEnvironment,
+                   String appCommandMetricsEnvironmentVersion, Integer appCommandGsUtil,
+                   File appCommandCredentialFile, String appCommandOutputFormat, boolean async,
+                   List<ProcessOutputLineListener> stdOutLineListeners,
+                   List<ProcessOutputLineListener> stdErrLineListeners,
+                   ProcessExitListener exitListener,
+                   int runDevAppServerWaitSeconds) {
```

Comment:
I guess this is the best we can do combing all methods into one.  I remember you talking about separating everything out (by command), we should think about how to do that next.